src/thing_creature.c

```diff
@@ -3830,12 +3830,12 @@ long player_list_creature_filter_dragging_specific_thing(const struct Thing *thi
         }
         return -1;
     }
-    if ((param->class_id > 0) || (param->model_id > 0) || (param->plyr_idx >= 0))
+    if ((param->class_id > 0) || ((param->model_id > 0) && (!is_creature_model_wildcard(param->model_id) && param->class_id == TCls_Creature)) || (param->plyr_idx >= 0))
```

Comment:
It always used as `class_id == TCls_Creature` and `model_id == ANY`
So we could replace and rename whole function to `find_any_creature_dragging_specfic_thing` and remove unused cases